examples/spl/token-proxy/programs/token-proxy/src/lib.rs

```diff
@@ -18,6 +18,32 @@ mod token_proxy {
     pub fn proxy_burn(ctx: Context<ProxyBurn>, amount: u64) -> ProgramResult {
         token::burn(ctx.accounts.into(), amount)
     }
+
+    pub fn proxy_set_authority(
+        ctx: Context<ProxySetAuthority>,
+        authority_type: AuthorityType,
+        new_authority: Option<Pubkey>,
+    ) -> ProgramResult {
+        let spl_authority_type = match authority_type {
```

Comment:
We should implement `impl From<AuthorityType> for spl_token::instruction::AuthorityType` so that we can hide this match statement.